docs/source/external-structures.rst

```diff
@@ -271,7 +271,7 @@ The following definitions are required:
 .. code::
 
     template <class T>
-    struct xcontainer_inner_type<table_adaptor<T>>
+    struct xcontainer_inner_types<table<T>>
```

Comment:
`xcontainer_inner_type<table_adaptor<T>>` was correct, you want to define inner types for the expression adaptor (i.e. `table_adaptor`), not for the adaptee (i.e. `table`).

Reply:
:+1:  but there was still a typo (`xcontainer_inner_types`).

Reply:
Ah indeed, good catch.